File: 0448_FindAllNumbersDisapperedInAnArray/Hank/448.py
class Solution:
    def findDisappearedNumbers(self, nums: List[int]) -> List[int]:
        length = len(nums)
        nums = set(nums)
        lost = []
        for i in range(1, length + 1):
            if i not in nums:
                lost.append(i)
        return lost

class Solution:
    def findDisappearedNumbers(self, nums: List[int]) -> List[int]:
        for i in range(len(nums)):
            index = abs(nums[i]) - 1
            nums[index] = -1 * abs(nums[index])
        return [i+1 for i in range(len(nums)) if nums[i] > 0]

File: 0448_FindAllNumbersDisapperedInAnArray/Jason/448.go
package main

func main() {

}

func FindDisappearedNumbers(nums []int) []int {
	// Version 1
	for _, num := range nums {
		if num < 0 {
			num *= -1
		}

		if nums[num-1] > 0 {
			nums[num-1] *= -1
		}
	}

	result := []int{}
	for i, num := range nums {
		if num > 0 {
			result = append(result, i+1)
		}
	}

	return result
}
